src/AppInstallerCommonCore/AppInstallerStrings.cpp

```diff
@@ -127,29 +155,27 @@ namespace AppInstaller::Utility
     {
         ICUBreakIterator itr{ input, UBRK_CHARACTER };
 
-        size_t utf8Offset = 0;
+        // Offset was past end, throw just like std::string::substr
+        if (itr.Advance(offset) == UBRK_DONE)
+        {
+            throw std::out_of_range("UTF8Substring: offset past end of input");
+        }
+
+        size_t utf8Offset = itr.CurrentOffset();
         size_t utf8Count = 0;
-        size_t graphemeClusterOffset = 0;
-        int32_t i = 0;
 
-        while (i != UBRK_DONE)
+        // Count past end, convert to npos to get all of string

```

Comment:
>Count past end [](start = 11, length = 14)

curious why do we not treat this as a bad input?

Reply:
For consistency with std::string::substr

---
In reply to: [463899310](https://github.com/Microsoft/pkg/pull/521#discussion_r463899310) [](ancestors = 463899310)